metaflow/plugins/argo/argo_client.py

```diff
@@ -52,6 +52,35 @@ def get_workflow_template(self, name):
                 json.loads(e.body)["message"] if e.body is not None else e.reason
             )
 
+    def get_workflow_templates(self, flow_name=None):
+        """
+        Return a list of workflow template names for a given Flow name.
+        Returns all workflow template names if no name is provided
+        """
+        client = self._client.get()
+        try:
+            results = client.CustomObjectsApi().list_namespaced_custom_object(
+                group=self._group,
+                version=self._version,
+                namespace=self._namespace,
+                plural="workflowtemplates",
+            )["items"]
+            return [
+                result["metadata"]["name"]
+                for result in results
+                if flow_name is None
+                or flow_name
+                == result["metadata"]
+                .get("annotations", {})
```

Comment:
is it possible to push this filter server-side? if we are only doing client-side filtering, then we can simply have this filtering logic in `argo_workflows.py`

Reply:
unfortunately not possible to push this server-side, as annotations are not supported for `field_selectors` for argo.

I also wanted to do the trimming of the return in the client as early as possible, as the API Response contains the full workflow templates, which can be quite hefty. Can move the filtering to `argo_workflows.py` too if it makes more sense.